cloudSupport/src/main/scala/cromwell/cloudsupport/aws/auth/AwsAuthMode.scala

```diff
@@ -91,85 +92,92 @@ sealed trait AwsAuthMode {
 case object MockAuthMode extends AwsAuthMode {
   override val name = "no_auth"
 
-  lazy val _credential = AnonymousCredentialsProvider.create.resolveCredentials()
-
-  override def credential(options: OptionLookup): AwsCredentials = _credential
+  override def provider(): AwsCredentialsProvider = AnonymousCredentialsProvider.create
 }
 
 object CustomKeyMode
 
+/**
+ * The AwsAuthMode constructed from a 'custom_key' auths scheme.
+ *
+ * @param name
+ * @param accessKey static AWS access key
+ * @param secretKey static AWS secret key
+ * @param region an optional AWS region
+ */
 final case class CustomKeyMode(override val name: String,
                                     accessKey: String,
                                     secretKey: String,
                                     region: Option[String]
                                     ) extends AwsAuthMode {
-  private lazy val _credential: AwsCredentials = {
+  private lazy val _provider: AwsCredentialsProvider = {
     // Validate credentials synchronously here, without retry.
     // It's very unlikely to fail as it should not happen more than a few times
     // (one for the engine and for each backend using it) per Cromwell instance.
-    validateCredential(AwsBasicCredentials.create(accessKey, secretKey), region)
+    StaticCredentialsProvider.create(validateCredential(AwsBasicCredentials.create(accessKey, secretKey), region))
   }
 
-  override def credential(options: OptionLookup): AwsCredentials = _credential
+  override def provider(): AwsCredentialsProvider = _provider
 }
 
+/**
+ * The AwsAuthMode constructed from a 'default' auths scheme.
+ *
+ * @param name
+ * @param region an optional AWS region
+ */
 final case class DefaultMode(override val name: String, region: Option[String]) extends AwsAuthMode {
-  private lazy val _credential: AwsCredentials = {
-    //
-    // The ProfileCredentialsProvider will return your [default]
-    // credential profile by reading from the credentials file located at
-    // (~/.aws/credentials).
-    //
-
-    // Validate credentials synchronously here, without retry.
-    // It's very unlikely to fail as it should not happen more than a few times
-    // (one for the engine and for each backend using it) per Cromwell instance.
-    validateCredential(DefaultCredentialsProvider.create.resolveCredentials(), region)
+  // The DefaultCredentialsProvider will look through a chain of standard AWS providers as
+  // per the normal behaviour of aws-cli etc
+  private lazy val _provider: AwsCredentialsProvider = {
+    DefaultCredentialsProvider.create()
```

Comment:
Wouldn't it make sense to modify `validateCredential` method to accept AwsCredentialsProvider, and use it to validate provider after creation? (here and in the other AwsAuthMode subclasses)

Reply:
I wasn't quite sure of the point of the validateCredential - given AWS itself will obviously validate credentials when they get used. My guess is that it is there as some sort of fail fast/region check - in which case materialising the credential immediately for that purpose is fine. But yes, am happy to change it to have the provider passed in.

Reply:
> My guess is that it is there as some sort of fail fast/region check - in which case materialising the credential immediately for that purpose is fine. But yes, am happy to change it to have the provider passed in.

I agree, looks like it was made this way in order to throw exception earlier in case if credentials are invalid. I think this behavior shouldn't be changed in scope of this PR, so it would be great if you could make an update.